kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/BaseConnectorNetworkIT.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright [2020 - 2020] Confluent Inc.
+ */
+
+package io.confluent.connect.s3.integration;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+
+import io.confluent.common.utils.IntegrationTest;
+import org.apache.kafka.test.TestUtils;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+@Category(IntegrationTest.class)
+public abstract class BaseConnectorNetworkIT extends BaseConnectorIT {
+
+  private static final Logger log = LoggerFactory.getLogger(BaseConnectorNetworkIT.class);
+
+  protected static final long CONSUME_MAX_DURATION_MS = TimeUnit.MINUTES.toMillis(20);
+  protected static final String JSON_FORMAT_CLASS = "io.confluent.connect.s3.format.json.JsonFormat";
+  protected static final String S3_BUCKET = "sink-test-bucket";
+
+  protected static PumbaPauseContainer pumbaPauseContainer;
+
+
+  /**
+   * Creates root client that will be used to change bucket permissions.
+   * Prerequisite : Access key and Secret access key should be set as environment variables
+   */
+  protected void createS3RootClient() {
+    log.info("Creating root S3 Client.");
+    S3Client = AmazonS3ClientBuilder.standard()
+        .withCredentials(
+            new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(
+                    System.getenv("ROOT_USER_ACCESS_KEY_ID"),
+                    System.getenv("ROOT_USER_SECRET_ACCESS_KEY"))))
+        .withRegion("ap-south-1")
+        .build();
+  }
+
+  @Override
+  protected long waitForFilesInBucket(
+      String bucketName,
+      int numFiles)
+      throws InterruptedException {
+    TestUtils.waitForCondition(
+        () -> assertFileCountInBucket(bucketName, numFiles).orElse(false),
+        CONSUME_MAX_DURATION_MS,
```

Comment:
What is the time consuming operation that requires us to have a 20 min timeout? How long do these tests take to run in total?

Reply:
The IT test `testWithNetworkInterruption()` use Pumba to cause interruptions in the connection with the s3 server, To verify these interruptions it is necessary for us to have a high record count in Kafka while having a small batch size. Therefore had set a high `CONSUME_MAX_DURATION_MS` timeout value.
The tests on the current configuration of 1000 records get executed within a couple of minutes barring `testWithNetworkInterruption()` which takes about 3-4 mins. So as per the current observation, I have reduced the `CONSUME_MAX_DURATION_MS` value to conservative  `8` minutes. Should we decrease it further?

Additionally

Reply:
Thanks for the clarification, 8 sounds much better.